slapo/schedule.py

```diff
@@ -428,6 +430,26 @@ def validate_sync_op(mode, sync_op_or_fn, axis=None):
                     sync_fn = partial(
                         reduce_scatter_forward_output, dim=axis, group=self.group
                     )
+                    # Avoid multiple reductions of linear bias
+                    # See https://github.com/awslabs/slapo/issues/10
+                    if (
+                        isinstance(self.mod, nn.Linear)
+                        and self.mod.bias is not None
+                        and self.metadata.shard["output_type"] == "partial"
+                    ):
+
+                        with init_empty_weights(
+                            enable=(self.mod.weight.device == torch.device("meta"))
+                        ):
+                            new_mod = LinearWithSeparateBias(
+                                self.mod.in_features,
+                                self.mod.out_features,
+                                sync_fn,
+                                self.mod.weight.device,
+                                self.mod.weight.dtype,
+                            )
+                            self.replace(new_mod)
+                        return
```

Comment:
Some concerns:
1. This breaks tie weights.
2. The execution flow of .shard becomes a bit difficult to flow, because here is an early return. I'm not sure if there's a better way but I'll think about it.

Reply:
Added `tie_weights` support